CHANGELOG.md

```diff
@@ -2,6 +2,263 @@
+* **Less boilerplate**: There’s no need to wrap these components manually in a `Fieldset` or manage duplicate legends.
+
+#### Action Required: Component-by-Component Guide
+
+##### Fieldset
+
+If you're using `Fieldset` with other form components, you will have to update your code with the following:
+
+* **Update usage:** Remove it if you’re using it to wrap the previous `Radio Group` and `Checkbox` components. The new `Radios` and `Checkboxes`  components come with a built-in `fieldset` now.
+
+* **Set legend size (required):** Use the new `legend-size` property to set the legend’s heading level from **H2** to **H6**, adjusting font size for proper hierarchy and accessibility.
+
+* **Remove validation:** `Fieldset` no longer handles validation, which was previously only used for radio and checkbox groups. Remove any usage of `validate-on` and `validator` on the component.
+
+* **Remove deprecated properties:** `fieldset-id`, `required`, `error-message`, and `disabled`. These were previously used only for validation.
+
+Here’s a code example to show the changes:
+
+###### Before:
+
+```html
+<gcds-fieldset
+  fieldset-id="field-required"
+  legend="Legend text"
+  hint="This is hint text."
+  required
+  error-message="This is an error message"
+>
+  // Grouped form components
+</gcds-fieldset>
+```
+
+###### After:
+
+```html
+<gcds-fieldset
+  legend="Legend text"
+  legend-size="h3"
+  hint="This is hint text."
+>
+  // Grouped form components
+</gcds-fieldset>
+```
+
+##### Radio Group to Radios
+
+Here’s how to change your `Radio Group` component usage to the new `Radios` component:
+
+* **Swap the component**: Replace `gcds-radio-group` with `gcds-radios`. It now includes a built-in `fieldset` and legend. If you haven’t done so yet, remove the `Fieldset` or `fieldset` wrapper around it.
+
+* **Use the legend property (required):** Define the group label using `legend`; manual group labeling is no longer needed.
+
+* **Access the selected value:** Use the `value` property to set and get the selected radio button’s value directly.
+
+* **New form properties:** Supports new properties like `required`, `hint`, `error-message`, `disabled`, `validator`, and `validate-on` for validation and user feedback.
+
+* **Remove manual validation wrappers:** Validation is handled internally; no need to wrap with `gcds-fieldset`.
+
+###### Before:
+
+```html
+<gcds-fieldset fieldset-id="radioFieldset" legend="Radio Options" hint="Choose one option." required="">
+<gcds-radio-group
+  name="radioDefault"
+  options='[{
+     "label": "Label for radio 1",
+     "id": "radio1",
+     "value": "radio1",
+     "hint": "Description or example to make the option clearer."
+   },{
+     "label": "Label for radio 2",
+     "id": "radio2", 
+     "value": "radio2",
+     "hint": "Description or example to make the option clearer."
+   }]'
+>
+</gcds-radio-group>
+</gcds-fieldset>
+
+```
```

Comment:
```suggestion
```html
<gcds-fieldset fieldset-id="radioFieldset" legend="Radio Options" hint="Choose one option." required="">
  <gcds-radio-group
    name="radioDefault"
    options='[{
       "label": "Label for radio 1",
       "id": "radio1",
       "value": "radio1",
       "hint": "Description or example to make the option clearer."
     },{
       "label": "Label for radio 2",
       "id": "radio2", 
       "value": "radio2",
       "hint": "Description or example to make the option clearer."
     }]'
  >
  </gcds-radio-group>
</gcds-fieldset>
```
```